src/components/views/rooms/BasicMessageComposer.tsx

```diff
@@ -515,6 +516,29 @@ export default class BasicMessageEditor extends React.Component<IProps, IState>
         }
     };
 
+    /**
+     * Because pills have contentEditable="false" there is no event emitted when
+     * the user tries to delete them. Therefore we need to fake what would
+     * normally happen
+     * @param direction in which to delete
+     * @returns handled
+     */
+    private fakeDeletion(direction: "deleteContentForward" | "deleteContentBackward" ): boolean {
```

Comment:
nit but `direction` is a bad name for this, IMO it should be a `backward: boolean` to simplify all the things. If you want to keep the string type then maybe `action`?

Reply:
and I guess the `"deleteContentBackward" : "deleteContentForward"` bit would move onto line 538

Reply:
Yeah looks much nicer

Reply:
also maybe `fakePillDeletion`

Reply:
No, not really, it is also used for deleting normal text as long as it isn't a range